src/app/dashboard/directory-dashboard/directory-items/directory-item/directory-item.component.ts

```diff
@@ -40,7 +40,9 @@ export class DirectoryItemComponent implements OnInit {
     this.directory.selected = false;
     this.updateDashboardDataSub = this.dashboardService.updateDashboardData.subscribe(val => {
       this.directory = this.directoryDbService.getById(this.directory.id);
-      this.directory.selected = false;
+      if(this.directory){
```

Comment:
the check for directory doesn't really do anything here. The line previous uses `this.directory.id` and a couple lines earlier has `this.directory.selected`